tasks/DroneFlightControlTask.cpp

```diff
@@ -76,38 +141,30 @@ void DroneFlightControlTask::updateHook()
 
     auto control_device =
         mVehicle->subscribe->getValue<Telemetry::TOPIC_CONTROL_DEVICE>();
+
+    States new_state = runtimeStatesTransition(control_device);
+    applyTransition(new_state);
+
+    if (state() != new_state)
+    {
+        state(new_state);
+    }
+
     mStatus.control_device = static_cast<ControlDevice>(control_device.deviceStatus);
     mStatus.device_flight_status =
         static_cast<DeviceFlightStatus>(control_device.flightStatus);
     _status.write(mStatus);
 
     // cmd input
     drone_dji_sdk::CommandAction cmd_action;
-    if (_cmd_action.read(cmd_action) == RTT::NoData)
+    if (_cmd_action.read(cmd_action) == RTT::NoData || state() == CONTROL_LOST)
```

Comment:
Probably it does not make a notable difference, but in:

```
if ( state() == CONTROL_LOST || _cmd_action.read(cmd_action) == RTT::NoData)
```
` _cmd_action.read` will be not performed if the first evaluation block fails (`state() == CONTROL_LOST`)

Not sure if this does fit on "earlier optimization"

Reply:
Wouldnt this only be true for `and` logical operations?

Reply:
Actually, for `or` operations it would stop evaluating if the condition is `true`, which in our case is the intention